Closes #1380

adam-hanna opened the pull request:
Per our [discussion](https://github.com/goadesign/goa/issues/1380). This should close #1380.

The tests that I wrote could be improved. For instance, I don't know how to wait for the server to start listening before the test suite continues.

raphael commented on service_test.go:
```diff
@@ -332,6 +336,81 @@ var _ = Describe("Service", func() {
+			ctrl := s.NewController("test")
+			s.Mux.Handle("GET", "/", ctrl.MuxHandler("get", handler, nil))
+
+			// note: should we be using an httptest / ghttp server, here?
```
Yes ghttp would indeed work well here - it would get rid of the race condition. https://godoc.org/github.com/onsi/gomega/ghttp has an example on how to use it also the [docs](https://onsi.github.io/gomega/#ghttp-testing-http-clients) are pretty good.

adam-hanna replied:
Hmm.. I'll have to think about this more. The problem is that `s.Server` is an `*http.Server`, not `*httptest.Server` nor `*ghttp.Server` so I can't assign those to it.

adam-hanna commented on service.go:
```diff
@@ -195,18 +200,23 @@ func (service *Service) LogError(msg string, keyvals ...interface{}) {
 	service.LogInfo("listen", "transport", "http", "addr", addr)
-	return http.ListenAndServe(addr, service.Mux)
+	service.Server.Addr = addr
+	service.Server.Handler = service.Mux
```
My only concern was that because because mux was assigned to `service.Server.Handler` during the `New` invocation, and that it wasn't a pointer, that any changes to `service.Mux` may not be available to `service.Server.Handler`. But I just took it out and it passed all tests so my concern seems unfounded.

raphael replied:
Yeah it's an interface which really is a reference type - not a value type even though there's no explicit pointer. Thanks for taking it out.

adam-hanna commented:
I removed the assignment of mux to `Service.Server.Handler` and refactored the tests. I didn't think it made sense to test that `Service.Server` behaved the way it was supposed to when passed parameters (e.g. that receiving a ReadTimeout would cause long-running requests to actually timeout). After all, that's what the http_test suite is for!

So, now just testing that `Service.Server` can receive props.

raphael commented:
Makes sense. The final version of the tests don't seem that useful either since all they are doing is making sure that a field they set is set... Seems like we should just get rid of the whole thing at this point and just leave the test that makes sure that `Server` is initialized when a service is created.

adam-hanna commented:
Agreed. Done.

raphael commented:
Great, thank you! one last thing :) do you think you could backport this change to v1:
1. Checkout the `v1` branch
2. Create a branch off of `v1`
3. Cherry-pick this commit into the branch
4. Create a new PR from the branch targetting `v1`
Thanks gain!
